dataset_loaders/videos/davis2017.py

```diff
@@ -0,0 +1,279 @@
+            binary one (only two ids 0-1).
+        dataset_version: string
+            A string in ['2016', '2017']. It allows to select the
+            version of Davis dataset to be loaded. Depending on this
+            parameter the files will be loaded from the most recent dataset
+            folder (2017) or from the previous one (2016).
+
+        """
+
+        if which_set not in {"train", "val", "test-dev"}:
+            raise ValueError("Unknown set {}".format(which_set))
+        self.which_set = which_set
+        self.threshold_masks = threshold_masks
+        self.multiobject = multiobject
+        if dataset_version not in ['2016', '2017']:
+            raise RuntimeError('Unknown dataset version')
+        self._image_sets_path = os.path.join(self.path, 'ImageSets',
+                                             dataset_version)
+        self.image_path = os.path.join(self.path,
+                                       'JPEGImages', '480p')
+        self.mask_path = os.path.join(self.path,
+                                      'Annotations', '480p')
+        self.rgb_values_shared_path = os.path.join(
+            self.shared_path, 'rgb_values_' + dataset_version + '.npy')
+        self.rgb_values_local_path = os.path.join(
+            self.path, 'rgb_values_' + dataset_version + '.npy')
+        if not os.path.exists(self.rgb_values_shared_path):
+            self.save_rgbs()
+
+        # Prepare data paths
+        self.split = 1. if self.which_set == 'test-dev' else split
+
+        super(Davis2017Dataset, self).__init__(*args, **kwargs)
+
+    def get_names(self):
+        """Return a dict of names, per prefix/subset."""
+        return self.filenames
+
+    def load_sequence(self, sequence):
+        """Load a sequence of images/frames
+
+        Auxiliary function that loads a sequence of frames with
+        the corresponding ground truth and their filenames.
+        Returns a dict with the images in [0, 1], their corresponding
+        labels, their subset (i.e. category, clip, prefix) and their
+        filenames.
+        """
+        from skimage import io
+
+        X = []
+        Y = []
+        F = []
+
+        rgbs = self.unique_rgbs
+        for prefix, frame_name in sequence:
+            img = io.imread(os.path.join(self.image_path, frame_name + '.jpg'))
+            img = img.astype(floatX) / 255.
+
+            if self.which_set in ['train', 'val']:
+                mask = np.asarray(Image.open(os.path.join(
+                    self.mask_path, frame_name + '.png')).convert('RGB'))
+            elif self.which_set == 'test-dev':
+                frame = self.filenames[prefix][0]
+                mask = np.asarray(Image.open(os.path.join(
+                    self.mask_path, frame + '.png')).convert('RGB'))
+            else:
+                raise RuntimeError()
+
+            # Convert mask from RGB to ids format
+            _id = 0
```

Comment:
Rather than using this ugly and anti-pythonic counter, why don't you store the id that comes out of the `mask.getcolors()` function into `self.unique_rgbs` and use it here? You can store id and rgb as a dictionary and iterate over it with iteritems(). Finally, when `self.foreground_background` is True you can just do `temp_mask[temp_mask > 1] = 1`

Reply:
Also, no need for temp_mask as I wrote you [here](https://github.com/fvisin/dataset_loaders/pull/9#discussion_r136697917) and [here](https://github.com/fvisin/dataset_loaders/pull/9#discussion_r136718642). Replace `mask[0]` with the id and at the end of the for loop replace `mask` with `mask[0]`.